src/endpoints/transactions/transaction.get.service.ts

```diff
@@ -211,4 +216,49 @@ export class TransactionGetService {
       return null;
     }
   }
+
+  async tryGetTransaction(txHash: string, fields?: string[]): Promise<TransactionDetailed | null> {
+    const txFromRedis = await this.cachingService.getCache<TransactionDetailed>(CacheInfo.Transaction(txHash).key);
+    if (txFromRedis) {
+      return txFromRedis;
+    }
+
+    const txFromDatabase = await this.persistenceService.getTransaction(txHash);
+    if (txFromDatabase) {
+      await this.cacheTransaction(txFromDatabase, { redis: true, database: false });
+      return txFromDatabase;
+    }
+
+    const txFromElastic = await this.tryGetTransactionFromElastic(txHash, fields);
+    if (txFromElastic) {
+      await this.cacheTransaction(txFromElastic);
+      return txFromElastic;
+    }
+
+    const txFromGateway = await this.tryGetTransactionFromGateway(txHash);
+    if (txFromGateway) {
+      await this.cacheTransaction(txFromGateway);
```

Comment:
I think we should not cache a transaction from gateway, because if a transaction is not yet indexed in elastic we will always return the gateway version for it, which is incomplete.